### DISCUSSION TITLE
What is React Strict Mode?

### DISCUSSION THREAD

**Discussion by super61803** [super9157/MARV-AI#8] [Q&A] [answered]
What is React Strict Mode?

**Comment by manoelduran** [comment, marked as answer]
React Strict Mode is a tool in React that helps identify potential issues in an application only during development. Its main goal is to help uncover problems in the code, making it more robust and ready for future React versions.

Key functions of Strict Mode:

1. Detect unexpected side effects, such as state mutations or unintended API calls;
2. Identify issues in the component lifecycle, performing extra checks on methods like componentDidMount and componentWillUnmount;
3. Detect dependency issues in Hooks, such as missing dependencies in useEffect;
4. Duplicate rendering of components (only in development) to identify non-idempotent code, meaning code that doesn't produce the same result in multiple renders.

You can learn more about how Strict Mode works in the official React documentation:

[doc](https://react.dev/reference/react/StrictMode)
